### ISSUE TITLE
POM rules report validation errors on the generated effective pom file instead of pom.xml

### ISSUE THREAD

**Issue by erangel** [avioconsulting/mule-linter#168]
With the new feature to support Maven Parent POM, POM rules run against the generated effective POM file.
When there is a validation error for any of POM Dependency, POM Plugin Attribute or POM Property Value or rules that inherit from them, the error is reported on the temporary file, instead of pom.xml
```
{
  "engineId": "The Munit maven plugin contains the required version. ",
  "ruleId": "MUNIT_PLUGIN_VERSION",
  "severity": "CRITICAL",
  "type": "CODE_SMELL",
  "primaryLocation": {
    "message": "Plugin exist but does not matches the attribute: version : 2.3.8",
    "filePath": "C:\\Users\\ELIAS~1.RAN\\AppData\\Local\\Temp\\effective-pom1729141150793602222.xml"
  }
},
{
  "engineId": "The munit.version maven property matches the given version. ",
  "ruleId": "MUNIT_VERSION",
  "severity": "CRITICAL",
  "type": "CODE_SMELL",
  "primaryLocation": {
    "message": "munit.version does not exist in \u003cproperties\u003e\u003c/properties\u003e",
    "filePath": "C:\\Users\\ELIAS~1.RAN\\AppData\\Local\\Temp\\effective-pom1729141150793602222.xml"
  }
}
```
When using the linter report with a tool like Sonarqube, these errors are ignored as they are not related to a file that exists in the code base.
Correct output for the previous sample should be:
```
{
  "engineId": "The mule maven plugin is a valid version. ",
  "ruleId": "MULE_MAVEN_PLUGIN",
  "severity": "CRITICAL",
  "type": "CODE_SMELL",
  "primaryLocation": {
    "message": "Plugin exist but does not matches the attribute: version : 3.3.5",
    "filePath": "pom.xml"
  }
},
{
  "engineId": "The munit.version maven property matches the given version. ",
  "ruleId": "MUNIT_VERSION",
  "severity": "CRITICAL",
  "type": "CODE_SMELL",
  "primaryLocation": {
    "message": "munit.version does not exist in \u003cproperties\u003e\u003c/properties\u003e",
    "filePath": "pom.xml"
  }
},
```
